implementation/src/main/java/com/teampc/controller/ViewSubmissionsController.java

```diff
@@ -17,27 +20,68 @@
 
    private Test currentTest;
 
+   @FXML
+   private Text testName;
+
+   @FXML
+   private Text className;
+
+   private String getTestDisplayStr(Test t) {
+      if (t == null) {
+         return "null Test";
+      }
+
+      String str = "";
+      if (t.getName() != null) {
+         str += t.getName();
+      }
+      if (t.getCourse() != null) {
+         str += " - " + t.getCourse();
+      }
+      return str;
+   }
+
    @Override
    public void initialize(URL location, ResourceBundle resources) {
       data = FXCollections.observableArrayList();
    }
 
+   // canned data
    private Submission s;
 
    // todo: alright, let's fix this later :(
    public void setCurrentTest(Test test) {
       this.currentTest = test;
 
-      this.s = new Submission();
-      this.s.responses.add(new MultipleChoiceQuestionResponse());
-      this.s.responses.add(new MultipleChoiceQuestionResponse());
-      this.s.taker = UserSession.loggedInUser;
+      testName.setText(getTestDisplayStr(currentTest));
+      className.setText(currentTest.getCourseName());
+
+      Collection<Submission> submissions = SubmissionDAO.getInstance().fetchSubmissionsForTest(currentTest);
+      Key key = currentTest.getKey();
+
+      // CANNED DATA
+      if (submissions.isEmpty()) {
+         Submission s = new Submission();
+         s.responses.add(new MultipleChoiceQuestionResponse());
+         s.responses.add(new MultipleChoiceQuestionResponse());
+         s.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
+         s.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
+         s.taker = UserSession.getLoggedInUser();
+
+         submissions = new ArrayList<>();
+         submissions.add(s);
+
+         key = new Key();
+         key.responses.add(new MultipleChoiceQuestionResponse());
+         key.responses.add(new MultipleChoiceQuestionResponse(1, new ArrayList<>()));
+         key.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
+         key.responses.add(new ShortAnswerQuestionResponse("one", ShortAnswerQuestionResponse.MatchType.ALL));
+      }
 
-      Collection<Submission> questionList = new ArrayList<Submission>();
-      questionList.add(s);
-      // SubmissionDAO.getInstance().fetchSubmissionsForTest(currentTest);
-      questionList.forEach(submission -> data.add(new RowView(new SubmissionRowController(submission.getTaker()), RESOURCE)));
+      final Key keyPC = key;
+      submissions.stream().forEach(nextSubmission -> nextSubmission.gradeTest(keyPC));
```

Comment:
fyi: collections have a forEach method on them now, so you don't need to create a stream just to foreach